### ISSUE TITLE
scoring callbacks fail with sklearn make_scorer

### ISSUE THREAD

**Issue by benjamin-work** [dnouri/skorch#173]
Using a scoring callback with `make_scorer` fails:

```
from sklearn.metrics import accuracy_score, make_scorer

net = NeuralNetClassifier(..., callbacks=[('accuracy', BatchScoring(make_scorer(accuracy)))])
```
The reason is that the name of the `make_scorer` object cannot be determined. To mitigate this, pass the `name` argument to the scoring callback.

This bug is especially annoying as users might believe it is possible to pass an sklearn metric as scoring function, which fails with the prompt to use `make_scorer`, which will then fail with the error above.
